src/render_tasks/d3d12_rt_shadow_task.hpp

```diff
@@ -0,0 +1,149 @@
+				auto& rs_registry = RootSignatureRegistry::Get();
+				data.base_data.out_root_signature = static_cast<D3D12RootSignature*>(rs_registry.Find(root_signatures::rt_hybrid_global))->m_native;
+			}
+
+			// Create Shader Tables
+			for (int i = 0; i < d3d12::settings::num_back_buffers; ++i)
+			{
+				CreateShaderTables(device, data.base_data, "ShadowRaygenEntry", i);
+			}
+
+			// Setup frame index
+			data.base_data.frame_idx = 0;
+		}
+
+		inline void ExecuteRTShadowTask(RenderSystem & render_system, FrameGraph & fg, SceneGraph & scene_graph, RenderTaskHandle & handle)
+		{
+			// Initialize variables
+			auto& n_render_system = static_cast<D3D12RenderSystem&>(render_system);
+			auto cmd_list = fg.GetCommandList<d3d12::CommandList>(handle);
+			auto& data = fg.GetData<RTShadowData>(handle);
+
+			Render(n_render_system, fg, scene_graph, data.base_data, cmd_list, handle, "ShadowRaygenEntry");
+			
+		}
+
+		inline void DestroyRTShadowTask(FrameGraph& fg, RenderTaskHandle handle, bool resize)
+		{
+			auto& data = fg.GetData<RTShadowData>(handle);
+
+			if (!resize)
+			{
+				// Small hack to force the allocations to go out of scope, which will tell the allocator to free them
+				std::move(data.base_data.out_output_alloc);
+				std::move(data.base_data.out_gbuffer_albedo_alloc);
+				std::move(data.base_data.out_gbuffer_normal_alloc);
+				std::move(data.base_data.out_gbuffer_depth_alloc);
+			}
+		}
+
+	} /* internal */
+
+	inline void AddRTShadowTask(FrameGraph& fg)
+	{
+		std::wstring name(L"Hybrid raytracing shadows");
+
+		RenderTargetProperties rt_properties
+		{
+			RenderTargetProperties::IsRenderWindow(false),
+			RenderTargetProperties::Width(std::nullopt),
+			RenderTargetProperties::Height(std::nullopt),
+			RenderTargetProperties::ExecuteResourceState(ResourceState::UNORDERED_ACCESS),
+			RenderTargetProperties::FinishedResourceState(ResourceState::COPY_SOURCE),
+			RenderTargetProperties::CreateDSVBuffer(false),
+			RenderTargetProperties::DSVFormat(Format::UNKNOWN),
+			RenderTargetProperties::RTVFormats({ Format::R16G16B16A16_FLOAT }),
+			RenderTargetProperties::NumRTVFormats(1),
+			RenderTargetProperties::Clear(true),
+			RenderTargetProperties::ClearDepth(true),
+			RenderTargetProperties::ResourceName(name),
+			RenderTargetProperties::ResolutionScalar(1.0f)
+		};
+
+		RenderTaskDesc desc;
+		desc.m_setup_func = [](RenderSystem& rs, FrameGraph& fg, RenderTaskHandle handle, bool resize)
+		{
+			internal::SetupRTShadowTask(rs, fg, handle, resize);
+		};
+		desc.m_execute_func = [](RenderSystem& rs, FrameGraph& fg, SceneGraph& sg, RenderTaskHandle handle)
+		{
+			internal::ExecuteRTShadowTask(rs, fg, sg, handle);
+		};
+		desc.m_destroy_func = [](FrameGraph& fg, RenderTaskHandle handle, bool resize)
+		{
+			internal::DestroyRTShadowTask(fg, handle, resize);
+		};
+
+		desc.m_properties = rt_properties;
+		desc.m_type = RenderTaskType::COMPUTE;
+		desc.m_allow_multithreading = true;
+
+		fg.AddTask<RTShadowData>(desc);
```

Comment:
This task has dependencies, please add them as the 2e parameter, see some other tasks for how.

Reply:
Added them. Added dependencies for other tasks I created as well.